examples/combined_ODNP.py

```diff
@@ -236,32 +241,48 @@
-        ph1_cyc=Ep_ph1_cyc,
-        adcOffset=config_dict["adc_offset"],
-        carrierFreq_MHz=config_dict["carrierFreq_MHz"],
-        nPoints=nPoints,
-        nEchoes=config_dict["nEchoes"],
-        p90_us=config_dict["p90_us"],
-        repetition_us=config_dict["repetition_us"],
-        tau_us=config_dict["tau_us"],
-        SW_kHz=config_dict["SW_kHz"],
-        indirect_fields=("start_times", "stop_times"),
-        ret_data=None,
-    )  # assume that the power axis is 1 longer than the
-    #                         "powers" array, so that we can also store the
-    #                         thermally polarized signal in this array (note
-    #                         that powers and other parameters are defined
-    #                         globally w/in the script, as this function is not
-    #                         designed to be moved outside the module
-    DNP_thermal_done = time.time()
-    time_axis_coords = DNP_data.getaxis("indirect")
-    time_axis_coords[0]["start_times"] = DNP_ini_time
-    time_axis_coords[0]["stop_times"] = DNP_thermal_done
+    for j in range(thermal_scans):
+        DNP_ini_time = time.time()
+        if j == 0:
+            DNP_data = run_spin_echo(
+                nScans=config_dict["nScans"],
+                indirect_idx=j,
+                indirect_len=len(powers) + thermal_scans,
+                ph1_cyc=Ep_ph1_cyc,
+                adcOffset=config_dict["adc_offset"],
+                carrierFreq_MHz=config_dict["carrierFreq_MHz"],
+                nPoints=nPoints,
+                nEchoes=config_dict["nEchoes"],
+                ph1_cyc=Ep_ph1_cyc,
+                p90_us=config_dict["p90_us"],
+                repetition_us=config_dict["repetition_us"],
+                tau_us=config_dict["tau_us"],
+                SW_kHz=config_dict["SW_kHz"],
+                indirect_fields=("start_times", "stop_times"),
+                ret_data=None,
+            )  
+            DNP_thermal_done = time.time()
+            time_axis_coords = DNP_data.getaxis("indirect")
+        else:
+            DNP_data = run_spin_echo(
+                nScans=parser_dict['nScans'],
+                indirect_idx=j,
+                indirect_len=len(powers) + thermal_scans,
+                adcOffset=parser_dict['adc_offset'],
+                carrierFreq_MHz=parser_dict['carrierFreq_MHz'],
+                nPoints=nPoints,
+                nEchoes=parser_dict['nEchoes'],
+                ph1_cyc=Ep_ph1_cyc,
+                p90_us=parser_dict['p90_us'],
+                repetition_us=parser_dict['repetition_us'],
+                tau_us=parser_dict['tau_us'],
+                SW_kHz=parser_dict['SW_kHz'],
+                indirect_fields=("start_times", "stop_times"),
+                ret_data=DNP_data,
+            )
+        DNP_thermal_done = time.time()
+        time_axis_coords[j]["start_times"] = DNP_ini_time
+        time_axis_coords[j]["stop_times"] = DNP_thermal_done
```

Comment:
Note to self to compare this to the code used elsewhere in the script. 

Also a question for Alex here.... What is the rationale where some of these are pulled directly from the dictionary and others of them are shuffled into variables first

Reply:
we can add Ep_ph1 and IR_ph1 etc to the config file but these are pretty much never going to change so I keep them defined in this script